packages/api/cms-api/src/generator/generate-crud.ts

```diff
@@ -67,7 +70,32 @@ function buildOptions(metadata: EntityMetadata<any>) {
     const hasSortArg = crudSortProps.length > 0;
 
     const hasSlugProp = metadata.props.some((prop) => prop.name == "slug");
-    const hasVisibleProp = metadata.props.some((prop) => prop.name == "visible");
+    let statusProp = metadata.props.find((prop) => prop.name == "status");
+    if (statusProp) {
+        if (!statusProp.enum) {
+            console.warn(`${metadata.className} status prop must be an enum to be supported by crud generator`);
+            statusProp = undefined;
+        } else if (statusProp.nullable) {
+            console.warn(`${metadata.className} status prop must not be nullable to be supported by crud generator`);
+            statusProp = undefined;
+        } else if (morphTsProperty(statusProp.name, metadata).getInitializer()?.getText() == "") {
+            console.warn(`${metadata.className} status prop must have a default value to be supported by crud generator`);
+            statusProp = undefined;
+        }
+    }
+    let statusActiveItems: Array<string | number> | undefined = undefined;
+    if (statusProp) {
+        if (!statusProp.items) throw new Error("Status enum prop has not items");
+        statusActiveItems = statusProp.items.filter((item) => {
+            if (typeof item == "number") {
+                console.warn(`${metadata.className} status prop must not have numeric items to be supported by crud generator`);
+                return false;
+            }
+            return ["Active", "Visible", "Invisible", "Published", "Unpublished"].includes(item);
+        });
+    }
+    const hasStatusFilter = statusProp && statusActiveItems && statusActiveItems.length != statusProp.items?.length; //if all items are active ones, no need for status filter
```

Comment:
I don't understand this: Why wouldn't I want to filter for e.g. unpublished items?

Reply:
Because we decided in our meeting:
- we want a archive and/or trash view in a dialog
- all others are considered as active and should be handled not in a dialog
- I designed the status filter to match that

Reply:
How would I query deleted/archived items for such a dialog?

Reply:
This is correct (or at least how I wanted it to be)

- Those values are grouped under "Active": "Active", "Visible", "Invisible", "Published", "Unpublished"
- they will be shown in a single view, filtering is possible using the standard filter api
- if no other stati exists (eg. Archived) the status filter will be omitted

It is also intended that products have no statusFilter (as Published/Unpublished are both "Active")

Reply:
I believe I know understand what you're doing. IMHO it would have been easier to understand if you'd also show the other use case in the PR. For instance, News could have a status active and deleted.

Reply:
done